src/modules/histogram.c

```diff
@@ -162,6 +162,7 @@ noit_log_histo_encoded_function_validate(noit_check_t *check, struct timeval *wh
   mtev_uuid_unparse_lower(check->checkid, uuid_str + strlen(uuid_str));
 
   unsigned long ms_cluster_jitter = noit_cluster_self_index() + 1;
+  if(explicit_time) ms_cluster_jitter = 0;
```

Comment:
We should probably initialize ms_cluster_jitter to 0, and then `if(!explicit_time) ms_cluster_jitter = noit_cluster_self_index() + 1;` to avoid making that call.   For a non-clustered broker, noit_cluster_self_index() is going to call mtev_cluster_by_name() every iteration, so I think it would be slightly faster in this case.